### ISSUE TITLE
USER STORY: Make a booking

### ISSUE THREAD

**Issue by HPCarey** [HPCarey/pt-booking#13]
As a **Site User** I can **click the book appointment button on the landing page and/or navbar** so that **I can fill in a form to book an appointment with the trainer.**

Acceptance Criteria:
- Authorised User can view and submit a booking  form

**Comment by HPCarey** [comment]
bug: Add and edit forms do not display django error message for empty field value for all fields. 
- Name fields and date and time fields don't display error mesage to user when left blank but other fields do.

**Troubleshooting Steps:**
1.  Although I assumed the fields would be required by default I did investigate this. I checked the source code and all fields have the requiredField class.
2.  I read up on some of the django model fields and I decided to adjust the fields to blank=False and migrate the changes to see if explicitly saying the fields are required would make a difference. Sources: [Stack Overflow](https://stackoverflow.com/questions/17199046/blank-null-and-required-in-django-models-and-modelforms), [Stack Overflow](https://stackoverflow.com/questions/48651166/validation-required-true-false-html-form), [Django docs](https://docs.djangoproject.com/en/4.2/ref/models/fields/)
3. Following from the above sources, here are some steps I took to try to ensure all fields were required and would throw an empty field error:
                           - I edited model field to explicitly have blank=False.
                           -  I tried to customise a specific empty field error using the clean_data method in forms.py. I did this to create other custom validation errors such as preventing users booking a date in the past with success, but for the empty field issue, I did not have success with this.
                           - Also in forms.py I tried using __init__ and set the fields to required but it didn’t work.

**Comment by HPCarey** [comment]
Fixed bug: 
- The django empty field error not displaying for empty field inputs for the first few values in the form was not a code error but a browser error. 
- This occurs in chrome and microsoft edge browsers (not firefox) when the form is longer than the viewport length.
-  The smooth scroll action to the unseen empty form field at the top of the page after the submit click causes the django empty field error not to be displayed.

Solution:
- The solution for this was shown to me by tutor support via this [stack overflow](https://stackoverflow.com/questions/69015407/html5-form-validation-message-doesnt-show-when-scroll-behaviour-is-set-to-smoo) article.
-  I set a rule in the css to override smooth scroll. 
- Now the viewport jumps to the empty form field instead of smooth scrolling up. The UX is not as pleasant but I decided to prioritise error handling in this instance.
-  The css code I used to fix this: 
 * {
    scroll-behavior: auto !important;
}
